src/Core/src/WeakEventHandler.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Maui
+{
+	/// <summary>
+	/// An internal, performance-focused version of WeakEventManager
+	/// By restricting to one event type, it can be strongly typed.
+	/// </summary>
+	class WeakEventHandler<TEventArgs> where TEventArgs : EventArgs
+	{
+		readonly List<WeakReference<EventHandler<TEventArgs>>> _subscriptions = new();
+
+		public void AddEventHandler(EventHandler<TEventArgs> handler)
+		{
+			if (handler is null)
+			{
+				throw new ArgumentNullException(nameof(handler));
+			}
```

Comment:
This is also `netstandard2.0`, otherwise I wanted to use `ArgumentNullExceptions.ThrowIfNull()`.